rest/data/creathost.go

```diff
@@ -0,0 +1,54 @@
+package data
+
+import (
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+// DBCreateHostConnector supports `create.host` commands from the agent.
+type DBCreateHostConnector struct{}
+
+// ListHostsForTask lists running hosts scoped to the task or the task's build.
+func (*DBCreateHostConnector) ListHostsForTask(taskID string) ([]model.CreateHost, error) {
+	t, err := task.FindOneId(taskID)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding task")
```

Comment:
return rest.APIError so that you can set the status code

Reply:
fixed